src/main/java/training/day5/School.java

```diff
@@ -0,0 +1,152 @@
+
+class School {
+    ArrayList<Student> students;
+    ArrayList<Teacher> teachers;
+
+    public School() {
+        students = new ArrayList<>();
+        teachers = new ArrayList<>();
+    }
+
+    public void addStudent(Student student) {
+        students.add(student);
+    }
+
+    public void addTeacher(Teacher teacher) {
+        teachers.add(teacher);
+    }
+
+
+    public void printInformation() {
+        System.out.println("Teachers:");
+        for (Teacher teacher : teachers) {
+            System.out.println(teacher);
+        }
+
+        System.out.println("Students:");
+        for (Student student : students) {
+            System.out.println(student);
+        }
+    }
+
+    public Student getStudentWithHighestGrade(String subject) {
+        Student studentWithHighestGrade = null;
+        double highestGrade = Double.MIN_VALUE;
+
+        for (Student student : students) {
+            Map<String, Double> grades = student.getGrades();
+            if (grades.containsKey(subject)) {
+                double grade = grades.get(subject);
+                if (grade > highestGrade) {
+                    highestGrade = grade;
+                    studentWithHighestGrade = student;
+                }
+            }
+        }
+
+        return studentWithHighestGrade;
+    }
+
+    public Student getStudentWithLowestGrade(String subject) {
+        Student studentWithLowestGrade = null;
+        double lowestGrade = Double.MAX_VALUE;
+
+        for (Student student : students) {
+            Map<String, Double> grades = student.getGrades();
+            if (grades.containsKey(subject)) {
+                double grade = grades.get(subject);
+                if (grade < lowestGrade) {
+                    lowestGrade = grade;
+                    studentWithLowestGrade = student;
+                }
+            }
+        }
+
+        return studentWithLowestGrade;
+    }
+
+    public double calculateAverageSalary() {
+        double totalSalary = 0;
+        for (Teacher teacher : teachers) {
+            totalSalary += teacher.getSalary();
+        }
+        return totalSalary / teachers.size();
+    }
+
+    public double calculateAverageGradeForStudent(Student student) {
+        double totalGrade = 0;
+        int gradeCount = 0;
+
+        Map<String, Double> grades = student.getGrades();
+        for (double grade : grades.values()) {
+            totalGrade += grade;
+            gradeCount++;
+        }
+
+        if (gradeCount > 0)
+            return totalGrade / gradeCount;
+        return 0;
+    }
+
+    public double calculateAverageGradeForClass() {
+        double totalGrade = 0;
+        int gradeCount = 0;
+
+        for (Student student : students) {
+            Map<String, Double> grades = student.getGrades();
+            for (double grade : grades.values()) {
+                totalGrade += grade;
+                gradeCount++;
+            }
+        }
+
+        if (gradeCount > 0)
```

Comment:
This is good. On a syntactic point to view, you can make the if statement with brackets because right after it you have another "return". It would be more readable.